lib/src/dart2_suggestors/pubspec_over_react_upgrader.dart

```diff
@@ -34,58 +37,66 @@ class PubspecOverReactUpgrader implements Suggestor {
 
-  final VersionRange targetConstraint;
+  /// Whether or not the codemod should add `over_react` if it is not already
+  /// found.
+  final bool shouldAddDependencies;
 
-  PubspecOverReactUpgrader(this.targetConstraint);
+  PubspecOverReactUpgrader(this.targetConstraint,
+      {this.shouldAddDependencies = true})
+      : shouldIgnoreMin = false;
+
+  /// Constructor used to ignore checks and ensure that the codemod always
+  /// tries to update the constraint.
+  ///
+  /// This is useful because the codemod may want to enforce a specific
+  /// range, rather than a target upper or lower bound. The only time this
+  /// will not update the pubspec is if the target version range is equal to
+  /// the version that is already there (avoiding an empty patch error).
+  PubspecOverReactUpgrader.alwaysUpdate(this.targetConstraint,
+      {this.shouldAddDependencies = true})
+      : shouldIgnoreMin = true;
 
   @override
   Iterable<Patch> generatePatches(SourceFile sourceFile) sync* {
     final contents = sourceFile.getText(0);
-    final overReactMatch = overReactDep.firstMatch(contents);
+    final overReactMatch = overReactDependencyRegExp.firstMatch(contents);
     if (overReactMatch != null) {
       // over_react is already in pubspec.yaml
-      final line = overReactMatch.group(0);
-      final constraintValue = overReactMatch.group(1);
+      final constraintValue = overReactMatch.group(2);
       final constraint = VersionConstraint.parse(constraintValue);
-      if (constraint is VersionRange && constraint.min < targetConstraint.min) {
+
+      if (shouldUpdateVersionRange(
+          targetConstraint: targetConstraint,
+          constraint: constraint,
+          shouldIgnoreMin: shouldIgnoreMin)) {
         // Wrap the new constraint in quotes if required.
         var newValue = targetConstraint.toString();
-        if (newValue.contains(' ') &&
-            !line.contains("'") &&
-            !line.contains('"')) {
-          newValue = "'$newValue'";
+        if (mightNeedYamlEscaping(newValue)) {
+          newValue = '"$newValue"';
         }
 
         // Update the version constraint to ensure a safe minimum bound.
         yield Patch(
           sourceFile,
           sourceFile.span(overReactMatch.start, overReactMatch.end),
-          line.replaceFirst(
-            constraintValue,
-            newValue,
-          ),
+          'over_react: $newValue',
         );
       }
-    } else {
+    } else if (shouldAddDependencies) {
       // over_react is missing in pubspec.yaml, so add it.
-      final dependeniesKeyMatch = dependenciesKey.firstMatch(contents);
+      final dependeniesKeyMatch = dependencyRegExp.firstMatch(contents);
       if (dependeniesKeyMatch != null) {
         // Wrap the new constraint in quotes if required.
         var newValue = targetConstraint.toString();
         if (newValue.contains(' ')) {
-          newValue = "'$newValue'";
+          newValue = '"$newValue"';
```

Comment:
#nit Any reason the quote type was changed? (same as above)

Reply:
Yep! Keal mentioned a bit back that we should default to double quotes if it was convenient, so I just swapped it out.

Reply:
:butwhy:

Reply:
Offline: pubspecs in the wild more often than not use double quotes